command/util.go

```diff
@@ -0,0 +1,75 @@
+package command
+
+import (
+	"fmt"
+	urlpkg "net/url"
+
+	"github.com/peak/s5cmd/storage"
+	"github.com/peak/s5cmd/storage/url"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	versioningNotSupportedWarning = "versioning related features are not supported with the given endpoint %q"
+	allVersionsFlagName           = "all-versions"
+	versionIDFlagName             = "version-id"
+)
+
+// checkVersinoningURLRemote checks if the versioning related flags are used with
+// local objects. Because the versioning is only supported with s3.
+func checkVersinoningURLRemote(url *url.URL) error {
+	if !url.IsRemote() && url.IsVersioned() {
+		return fmt.Errorf("%q, and %q flags can only be used with remote objects", allVersionsFlagName, versionIDFlagName)
+	}
+	return nil
+}
+
+// checkVersioningFlagCompatibility checks if the incompatible versioning flags
+// are used together. Because it is not allowed to refer to both "all versions" and
+// a specific version of an object together.
+func checkVersioningFlagCompatibility(ctx *cli.Context) error {
+	if ctx.Bool(allVersionsFlagName) && ctx.String(versionIDFlagName) != "" {
+		return fmt.Errorf("it is not allowed to combine %q and %q flags", allVersionsFlagName, versionIDFlagName)
+	}
+	return nil
+}
+
+// checkVersioningWithGoogleEndpoint checks if the versioning flags are used with
+// the Google Endpoint. Because the s3 versioning operations are not compatible with
+// GCS's versioning API.
+func checkVersioningWithGoogleEndpoint(ctx *cli.Context) error {
+	endpoint := ctx.String("endpoint-url")
+	if endpoint == "" {
+		return nil
+	}
+
+	u, err := urlpkg.Parse(endpoint)
+	if err != nil {
+		return err
+	}
+
+	if storage.IsGoogleEndpoint(*u) && (ctx.Bool(allVersionsFlagName) || ctx.String(versionIDFlagName) != "") {
+		return fmt.Errorf(versioningNotSupportedWarning, endpoint)
+	}
+
+	return nil
+}
+
+// checkNumberOfArguments checks if the number of the arguments is valid.
+// if the max is negative then there is no upper limit of arguments.
+func checkNumberOfArguments(ctx *cli.Context, min, max int) error {
+	l := ctx.Args().Len()
+	if min == 1 && max == 1 && l != 1 {
+		return fmt.Errorf("expected only one argument")
+	}
+	if min == 2 && max == 2 && l != 2 {
```

Comment:
as far as I can see this function is only called on [bucket_version.go](https://github.com/Kucukaslan/s5cmd/blob/d48b45b2596233e37411182624487d2cc27034e0/command/bucket_version.go#L52) and its min/max parameters are always 1/1. Do we still need to check other values (for example 2/2 in this line)?

Reply:
No but yes too. I intentionally made it general function. Becuase I hope it can be used by all commands in future. But I didn't change other commands' validations checks, since they would be rather irrelevant to this PR.

Reply:
👍 it makes sense now.